src/main/java/co/cask/PluginJSON.java

```diff
@@ -196,6 +204,73 @@ public void execute()  throws MojoExecutionException {
   }
 
   /**
+   * Inspects all the Icon files to add properties to related widget.
+   */
+  private void addIcon(JSONObject finalObject, Map<String, File> files, File nameFile) throws IOException {
```

Comment:
lets do a little cleanup here, starting with using more descriptive names. 'finalObject' -> 'widgetsJson', 'files' -> 'iconFiles'.

Also shouldn't be passing in the 'nameFile', as you're not using the file at all and it's already been read. Just pass in the stripped filename as a string, especially since the caller strips it anyway in the next line.